src/PactNet.AspNetCore/Messaging/MessageMiddleware.cs

```diff
@@ -0,0 +1,151 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using PactNet.AspNetCore.Messaging.Models;
+using PactNet.AspNetCore.Messaging.Options;
+using PactNet.Native;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace PactNet.AspNetCore.Messaging
+{
+    /// <summary>
+    /// Defines the message middleware
+    /// It handles the queries for the verifier when there are message interactions
+    /// </summary>
+    public class MessageMiddleware
+    {
+        /// <summary>
+        /// The default serializer options
+        /// </summary>
+        public static JsonSerializerOptions SerializerOptions => new JsonSerializerOptions
+        {
+            PropertyNameCaseInsensitive = true
+        };
+
+        /// <summary>
+        /// the list of scenarios
+        /// </summary>
+        private readonly Scenarios scenarios;
+
+        /// <summary>
+        /// The next pointer
+        /// </summary>
+        private readonly RequestDelegate next;
+
+        /// <summary>
+        /// The message middleware options
+        /// </summary>
+        private readonly MessagingVerifierOptions options;
+
+        /// <summary>
+        /// Creates an instance of <see cref="MessageMiddleware"/>
+        /// </summary>
+        /// <param name="options">the middleware options</param>
+        /// <param name="scenarios">the list of scenarios</param>
+        /// <param name="next">the next handle</param>
+        public MessageMiddleware(IOptionsMonitor<MessagingVerifierOptions> options, Scenarios scenarios, RequestDelegate next)
```

Comment:
That just needs to take an `IOptions` doesn't it? We're not expecting the options to change one the test is running.